src/expr-visitor.cc

```diff
@@ -61,14 +61,24 @@ Result ExprVisitor::VisitExpr(Expr* expr) {
       break;
 
     case ExprType::Catch:
-      // TODO(karlschimpf): Define
-      WABT_FATAL("Catch: don't know how to visit\n");
-      return Result::Error;
+      if (!CommonClOptions.allow_exceptions) {
```

Comment:
I changed the philosophy to assume that the reader/parser checks if these constructs are allowed, and removed the checks in the code.

I also reorganized the model for catches, and they are no longer an Expr.

The only two conditions that the validator now needs to check are:
1) Are all rethrow instructions in a catch block.
2) The catch all (if it appears) must be the last catch.

I also wrapped catch and catch_all into a single construct (with an invalid index representing a catch all).